ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ConvertTupleToElementsFix.fs

```diff
@@ -0,0 +1,76 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open System.Text.RegularExpressions
+open JetBrains.ReSharper.Feature.Services.Intentions.Scoped
+open JetBrains.ReSharper.Feature.Services.Intentions.Scoped.Actions
+open JetBrains.ReSharper.Feature.Services.Intentions.Scoped.Scopes
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Util
+open JetBrains.ReSharper.Resources.Shell
+
+module ConvertTupleToElements =
+    let regex = Regex("'\w\d*( \* '\w\d*)+")
+
+    let getArrayOrListParent (expr: IFSharpExpression) =
+        let expr = expr.IgnoreParentParens()
+        let arrayOrList = ArrayOrListExprNavigator.GetByExpression(expr)
+        let arrayOrList =
+            if isNotNull arrayOrList then arrayOrList else
+                let seq = SequentialExprNavigator.GetByExpression(expr)
+                if isNotNull seq then ArrayOrListExprNavigator.GetByExpression(seq) else null
+        arrayOrList
+
+    let isAvailable expr actual =
+        isValid expr && isNotNull (getArrayOrListParent expr) && regex.Match(actual).Value = actual
+
+    let convertTupleToElements (expr: IFSharpExpression) (factory: IFSharpElementFactory) =
+        match expr with
+        | :? ITupleExpr as tuple ->
+            let sequentialExpr = factory.CreateSequentialExpr(tuple.ExpressionsEnumerable)
```

Comment:
Could you check this approach on samples like:

Split over multiple lines:
```fsharp
[ 1, 2,
  3, 4 ]
```

Comments:
```fsharp
[ 1,

  // a very important comment here
  2 ]
```

I'd try creating the sequence expression via `ElementType.SEQUENTIAL_EXPR.Create()` and then moving the children nodes there, if needed.

Since the quick fix is also available in cases where the tuple is not the only list element (`[1, 2; 3]`), you have to make sure you don't get nested sequence expressions. Speaking about it, I'm not yet convinced we need this variant of the quick fix (at least now, when types info we have is somewhat limited here), since the main idea was about fixing wrong separators for the whole lists and not about fixing inner expressions in lists.